### ISSUE TITLE
deliver to central

### ISSUE THREAD

**Issue by McFoggy** [bdemers/maven-external-version#3]
Would it be possible that the project is deployed to maven central?

I know it introduces some pain for the project responsible but it facilitates adoption and reuse.

**Comment by McFoggy** [comment]
@bdemers do you plan to consider this delivery to central? I have finished the jgit plugin but I need to know what you want to do with your own module before releasing mine.
Alternatives will be from fork to rewrite and host behind my own umbrella for delivery to central (http://oss.brouillard.fr) ; but honestly none of those solutions satisfy me.

**Comment by bdemers** [comment]
Yes, if it is at t usable level, your changes in #2 might be enough to make it useable.  Can you point me to your jgit plugin so I can see how you are using this?

**Comment by McFoggy** [comment]
I have not yet published it. If you have read my [blog post](http://blog.matthieu.brouillard.fr/2016/04/14/version-handling-with-git-and-maven/) I think you have an idea how it can work if you replace `git describe` with some possibilities offered by direct JGit integration. I need to document a little bit what I did ; but what I can tell you is that I make use of the differences between annotated tags & lightweight ones in order to derive project versions from git commits/tags.

**Comment by gerardbosch** [comment]
Hi there! I find these 2 plugins @bdemers , @McFoggy very interesting and I would like to consume it in my organization. I see it's 3 years since it was created but I can't found it on Maven Central.

Any hopes your work can be published there? How can I use it if not? I guess I could clone it and build it inside my organization so it would become deployed at our binary repository (Artifactory), but maybe this is not the best approach...

I'm very interested in the feature of dynamically computing the POM version from the output of a shell script run (strategy script I think).

Really cool, thanks!

**Comment by bdemers** [comment]
Hey @gerardbosch!

I haven't had time to work on this (and then either added back to Apache Maven, or change the package/group).

I'd love to pick this up again, but I'm not sure how much interest there is in this plugin (and how much development help I'd get).

Does the version in this repo work for you as is?

**Comment by gerardbosch** [comment]
Hi there, I haven't tried it yet but I will try it during the week and I will back to you.

I'm also very interested in @McFoggy strategy jgitvers to automatically update versions based on git history.

I've never tried the maven-release-plugin but it looks a bit cumbersome and complex.

**Comment by McFoggy** [comment]
@gerardbosch 
several years ago I built a plugin [maven-external-version-jgitver](https://github.com/jgitver/maven-external-version-jgitver) for @bdemers maven-external-version.

Then I switched my mind and provided an autonomous maven core extension that computes automatically the projects version, it is called [jgitver-maven-plugin](https://github.com/jgitver/jgitver-maven-plugin) and is based on [jgitver](https://github.com/jgitver/jgitver) which effectively does the version computation.

with [jgitver-maven-plugin](https://github.com/jgitver/jgitver-maven-plugin) my release process is as simple as:
- `git tag`
- `mvn deploy`

**Comment by bdemers** [comment]
Very cool!! I'll have to check that out!

**Comment by gerardbosch** [comment]
Hey @McFoggy , I've found it **very** interesting. I will definitely try to introduce it in my organization's project workflow. I really hope it can fit with the DevOps model defined in the company I work for, because it is not very well designed and it is quite rigid. But as I can see, it looks everything happens in the validate phase, so it may work out for me!

Sure I will try it!! Really nice work, thanks!


Regarding the use case I wanted to cover originally (computing the version from the output of a shell script), I managed to make it through the `it.ozimov:yaml-properties-maven-plugin` (that reads properties from a YAML) and `versions-maven-plugin`; as what I pretended was to take the POM version from a YAML file.

That worked for now with a little of hackery on the clean lifecycle through these plugins. But I will probably drop this approach and try to build on a Gradle or SBT solution (I think at least Gradle can be run from Maven) as I think it should be cleaner and more straightforward.

**Comment by whatthefrog** [comment]
playing a bit with jgitver-maven-plugin 
I guess it should be mentioned in this plugin README

Thanks for the good work @bdemers and @McFoggy
